x-pack/plugins/enterprise_search/public/applications/analytics/components/analytics_collection_view/analytics_collection_data_view_logic.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { DataView } from '@kbn/data-views-plugin/common';
+
+import { KibanaLogic } from '../../../shared/kibana/kibana_logic';
+
+import { FetchAnalyticsCollectionLogic } from './fetch_analytics_collection_logic';
+
+interface AnalyticsCollectionDataViewLogicValues {
+  dataView: DataView | null;
+}
+
+interface AnalyticsCollectionDataViewLogicActions {
+  setDataView(dataView: DataView): { dataView: DataView };
+}
+
+export const AnalyticsCollectionDataViewLogic = kea<
+  MakeLogicType<AnalyticsCollectionDataViewLogicValues, AnalyticsCollectionDataViewLogicActions>
+>({
+  actions: {
+    setDataView: (dataView) => ({ dataView }),
+  },
+  listeners: ({ actions }) => ({
+    [FetchAnalyticsCollectionLogic.actionTypes.apiSuccess]: async (collection) => {
+      const dataView = (
```

Comment:
nice! I wonder if we could do here is create the dataview if doesn't exist too (on a future PR). Related to the create dataview if doesn't exist issue.

Reply:
Yep, Klim working on that, I told him about this new logic, so he probably will use it